jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -114,6 +118,38 @@ public static boolean isCRIUSupportEnabled() {
 		return criuSupportEnabled;
 	}
 
+	private String checkPointDir;
+	private boolean keepRunning = false;
+	private boolean shellJob = true;
+	private boolean extUnixSupport = true;
+	private int logLevel = 4;
+	private String logFile = "checkpoint.log";
+
+	public CRIUSupport setCheckPointDir(String checkPointDir) {
+		this.checkPointDir = checkPointDir;
+		return this;
+	}
+	public CRIUSupport setKeepRunning(boolean keepRunning) {
+		this.keepRunning = keepRunning;
+		return this;
+	}
+	public CRIUSupport setShellJob(boolean shellJob) {
+		this.shellJob = shellJob;
+		return this;
+	}
+	public CRIUSupport setExtUnixSupport(boolean extUnixSupport) {
+		this.extUnixSupport = extUnixSupport;
+		return this;
+	}
+	public CRIUSupport setLogLevel(int logLevel) {
+		this.logLevel = logLevel;
+		return this;
+	}
+	public CRIUSupport setLogFile(String logFile) {
```

Comment:
same here, fileName can be a string

Reply:
Do you mean Path?

Reply:
I was thinking `setLogFile(Path path, String fileName)`, @keithc-ca any thoughts?

Reply:
Why two arguments? Why not just expect `path` to refer to a file to receive the log output?

Reply:
Also I am getting this error
```
Error (criu/cr-service.c:390): No subdirs are allowed in log_file name: No such file or directory
res.getType() SYSTEM_CHECKPOINT_FAILURE
java.lang.InternalError: Could not dump the JVM processs, err=-52
        at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkpointJVMImpl(Native Method)
        at openj9.criu/org.eclipse.openj9.criu.CRIUSupport.checkPointJVM(CRIUSupport.java:210)
        at Hello.main(Hello.java:14)
```
for this method:
```java
public CRIUSupport setLogFile(Path logFile) {
	if (logFile != null) {
		String file = logFile.toString();
		SecurityManager manager = System.getSecurityManager();
		if (manager != null) {
			manager.checkPermission(CRIU_DUMP_PERMISSION);
			manager.checkWrite(dir);
		}
		this.logFile = dir;
	}
	return this;
}
```
Is it possible that the log_file can only be the name of a file and we can't specify a directory?

Reply:
What was your input?

Reply:
`.setLogFile(Paths.get("checkpoint.log"))`

Reply:
Yup, its not documented but logfile will always be in the imageDir. So we should only supply a filename

Reply:
Perhaps internally the given `Path` needs to be decomposed into a directory and a simple filename.

Reply:
we should validate that a path is not specified so we dont hit the error

Reply:
> Perhaps internally the given Path needs to be decomposed into a directory and a simple filename.

The logfile is stored in the workDir (defaults to imageDir) but can be set with `criu_local_set_work_dir_fd`. So we will need to add that.